File: schemaorg-rs/src/organization_role.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/endDate
///https://schema.org/Date
///https://schema.org/DateTime
pub type OrganizationRoleEndDateFieldEnum = String;
///https://schema.org/startDate
///https://schema.org/Date
///https://schema.org/DateTime
pub type OrganizationRoleStartDateFieldEnum = String;
///https://schema.org/roleName
///https://schema.org/Text
///https://schema.org/URL
pub type OrganizationRoleRoleNameFieldEnum = String;
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type OrganizationRoleAdditionalTypeFieldEnum = String;
///https://schema.org/namedPosition
///https://schema.org/Text
///https://schema.org/URL
pub type OrganizationRoleNamedPositionFieldEnum = String;
///https://schema.org/OrganizationRole
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct OrganizationRole {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OrganizationRoleDescriptionFieldEnum>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OrganizationRoleImageFieldEnum>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OrganizationRoleMainEntityOfPageFieldEnum>,
    ///https://schema.org/endDate
    #[serde(rename = "endDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub end_date: Vec<OrganizationRoleEndDateFieldEnum>,
    ///https://schema.org/startDate
    #[serde(rename = "startDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub start_date: Vec<OrganizationRoleStartDateFieldEnum>,
    ///https://schema.org/roleName
    #[serde(rename = "roleName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub role_name: Vec<OrganizationRoleRoleNameFieldEnum>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OrganizationRoleAdditionalTypeFieldEnum>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OrganizationRoleIdentifierFieldEnum>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OrganizationRoleSubjectOfFieldEnum>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/numberedPosition
    #[serde(rename = "numberedPosition")]
    #[serde_as(as = "OneOrMany<_>")]
    pub numbered_position: Vec<f32>,
    ///https://schema.org/namedPosition
    #[serde(rename = "namedPosition")]
    #[serde_as(as = "OneOrMany<_>")]
    pub named_position: Vec<OrganizationRoleNamedPositionFieldEnum>,
}

File: schemaorg-ts/src/organization_role.ts
import type * as all from "./index";

/**
 * <https://schema.org/endDate>
 **/
export type OrganizationRoleEndDateFieldEnum = string;

/**
 * <https://schema.org/namedPosition>
 **/
export type OrganizationRoleNamedPositionFieldEnum = string;

/**
 * <https://schema.org/startDate>
 **/
export type OrganizationRoleStartDateFieldEnum = string;

/**
 * <https://schema.org/roleName>
 **/
export type OrganizationRoleRoleNameFieldEnum = string;

/**
 * <https://schema.org/additionalType>
 **/
export type OrganizationRoleAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/OrganizationRole>
 **/
export interface OrganizationRole {
  "@context": string;
  "@type": "OrganizationRole";

  /**
   * <https://schema.org/numberedPosition>
   **/
  numberedPosition: all.OneOrMany<number>;

  /**
   * <https://schema.org/endDate>
   **/
  endDate: all.OneOrMany<all.OrganizationRoleEndDateFieldEnum>;

  /**
   * <https://schema.org/namedPosition>
   **/
  namedPosition: all.OneOrMany<all.OrganizationRoleNamedPositionFieldEnum>;

  /**
   * <https://schema.org/startDate>
   **/
  startDate: all.OneOrMany<all.OrganizationRoleStartDateFieldEnum>;

  /**
   * <https://schema.org/roleName>
   **/
  roleName: all.OneOrMany<all.OrganizationRoleRoleNameFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OrganizationRoleMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OrganizationRoleIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OrganizationRoleDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OrganizationRoleSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OrganizationRoleAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OrganizationRoleImageFieldEnum>;
}
